### PR TITLE
Update save data section in UG

### LINKED ISSUE #268: [PE-D][Tester A] Inaccurate documented behavior
The cautionary message is incorrect

> Caution: If your changes to the data file makes its format invalid, app will discard all data and start with an empty data file at the next run. Hence, it is recommended to take a backup of the file before editing it.

When an invalid edit is made, the data file is not wiped on the start of the next run. It is only overwritten when a command (e.g `list`) is executed. 

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: Oceankoh/ped#16

### PR DESCRIPTION
**Pull request by RezwanAhmed123**
The save data warning section is currently not consistent with the behaviour observed in the warning section of the UG with regards to corrupt data.

Let's update the UG to make it more consistent and not confuse users.

Fixes #268 .

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -337,16 +337,20 @@ App data are saved automatically as a JSON file `[JAR file location]/data/addres
 <box type="warning" seamless>
 
 **Caution:**
-If your changes to the data file makes its format invalid, app will discard all data and start with an empty data file at the next run.  Hence, it is recommended to take a backup of the file before editing it.<br>
-Furthermore, certain edits can cause the app to behave in unexpected ways (e.g., if a value entered is outside the acceptable range). Therefore, edit the data file only if you are confident that you can update it correctly.
+If your changes to the data file makes its format invalid, app will discard all data at the start of the next run. <br>
+A new data file with sample data will replace the corrupted file once a new successful command is executed.
```
Comment (yanqiyqh) [inline docs/UserGuide.md:341]: Actually I think that our app starts on a blank slate without any sample data, that was what I updated in the DG after testing. Maybe you can confirm on your side?

**Review by yanqiyqh — CHANGES_REQUESTED** [review]
Just one small issue

File: docs/UserGuide.md (hunk 1 above)
Comment (RezwanAhmed123) [reply docs/UserGuide.md:341]: you are right, good catch!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -337,16 +337,20 @@ App data are saved automatically as a JSON file `[JAR file location]/data/addres
 <box type="warning" seamless>
 
 **Caution:**
-If your changes to the data file makes its format invalid, app will discard all data and start with an empty data file at the next run.  Hence, it is recommended to take a backup of the file before editing it.<br>
-Furthermore, certain edits can cause the app to behave in unexpected ways (e.g., if a value entered is outside the acceptable range). Therefore, edit the data file only if you are confident that you can update it correctly.
+If your changes to the data file makes its format invalid, app will discard all data at the start of the next run. <br>
+An empty data file will replace the corrupted file once a new successful command is executed.
```
Comment (zi-yii) [inline docs/UserGuide.md:341]: Technically, the data file is not empty if the successful command is an add command. Would it be better to say "A new data file will replace ..."

**Review by zi-yii — COMMENTED** [review]
Just a small concern

File: docs/UserGuide.md (hunk 2 above)
Comment (RezwanAhmed123) [reply docs/UserGuide.md:341]: ok makes sense

**Review by zi-yii — APPROVED** [review]
LGTM

**Review by yanqiyqh — APPROVED** [review]
LGTM

**Pull request merged**